File: app/src/main/java/com/backyardev/spacejam/activities/SplashActivity.java
package com.backyardev.spacejam.activities;

import android.content.Intent;
import android.os.Bundle;

import com.backyardev.spacejam.R;
import com.google.firebase.auth.FirebaseAuth;

import androidx.appcompat.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {

    FirebaseAuth fAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate( savedInstanceState );
        setTheme( R.style.AppTheme_Splash );
        fAuth = FirebaseAuth.getInstance();
        if (fAuth.getCurrentUser() != null) {

            Intent i = new Intent( SplashActivity.this, HomeActivity.class );
            startActivity( i );
            finish();
        } else {
            Intent i = new Intent( SplashActivity.this, LoginActivity.class );
            startActivity( i );
            finish();
        }
    }
}

File: app/src/main/java/com/backyardev/spacejam/activities/SplashActivity.kt
package com.backyardev.spacejam.activities

import android.content.Intent
import android.os.Bundle

import com.backyardev.spacejam.R
import com.google.firebase.auth.FirebaseAuth

import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {

    private lateinit var fAuth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.AppTheme_Splash)
        fAuth = FirebaseAuth.getInstance()
        if (fAuth.currentUser != null) {
            val i = Intent(this@SplashActivity, HomeActivity::class.java)
            startActivity(i)
            finish()
        } else {
            val i = Intent(this@SplashActivity, LoginActivity::class.java)
            startActivity(i)
            finish()
        }
    }
}